### PR TITLE
feat(auth-server): convert `subscriptionAccountReminderFirst`

### PR DESCRIPTION
**Pull request by xlisachan**
Because:

* We are actively converting old FxA emails to our new modernized stack

This commit:

* Converts the `subscriptionAccountReminderFirst` subscription platform email to the new stack.

Closes: #10909

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)
Old template
Screen Shot 2021-12-20 at 9 52 23 AM

New template
Screen Shot 2021-12-20 at 9 52 37 AM

## Other information (Optional)
The formatting of this layout depends on the update to the button partial in #11367

I updated the button in the new email template to `Create Password` to match the text above which states "Select 'Create Password'....". If aligned, I'll add this change to the Confirmation Needed doc.

Additional updates include the following:

Revised the layout of the subplat email template to accommodate for reminder emails:
* `emails/layouts/subscription/en.ftl`
* `emails/layouts/subscription/index.mjml`
* `emails/layouts/subscription/index.stories.ts`
* `emails/layouts/subscription/index.txt `

Updated the subscription stylesheet for the `automatedEmailResetPassword` partial
* `emails/layouts/subscription/index.scss`
 
Nitpick: spelling error
* `fxa-auth-server/lib/senders/email.js`
* `test/local/senders/email.js`

Note, `subscriptionAccountReminderFirst` was already included in `dev.json`

### CODE REVIEW

File: packages/fxa-auth-server/lib/senders/emails/layouts/subscription/index.mjml [hunk 1]
```diff
@@ -54,34 +54,49 @@
               You’re receiving this email because <%- email %> has a Firefox account and you signed up for <%- productName %>.
             </span>
           <% } else { %>
-            <span data-l10n-id="subplat-explainer-multiple" data-l10n-args="<%= JSON.stringify({ email }) %>">
-              You’re receiving this email because <%- email %> has a Firefox account and you have subscribed to multiple products.
-            </span>
+            <% if (locals.reminderShortForm) { %>
```
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/layouts/subscription/index.mjml:57]: This is really minor but can we turn this into an `else if` instead?

File: packages/fxa-auth-server/lib/senders/emails/layouts/subscription/index.scss [hunk 2]
```diff
@@ -84,3 +84,18 @@
   width: 58px;
   height: 58px;
 }
+
+.text-footer div {
```
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/layouts/subscription/index.scss:88]: To eliminate repeating ourselves, we could add all of the common styles between subplat and fxa layouts into the global file.

In the global:
```
.text-footer div {
  @extend .font-sans;
  text-align: center !important;
  color: global.$grey-400 !important;
}

.text-footer-automatedEmail {
  @extend .text-footer;
  div {
    @extend .mb-3;
    @extend .mt-10;
  }
}
```
Then in fxa layout:

```
.text-footer div {
  @extend .text-xs;
}
```

and in subplat layout:
```
.text-footer div {
  @extend .text-sm;
}

File: packages/fxa-auth-server/lib/senders/emails/layouts/subscription/index.stories.ts [hunk 3]
```diff
@@ -17,6 +17,13 @@ const createStory = subplatStoryWithProps(
   }
 );
 
+export const LayoutNoProduct = createStory(
+  {
+    reminderShortForm: true,
+  },
+  'Reminder short form'
```
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/layouts/subscription/index.stories.ts:24]: What does this mean, short form? Like abbreviated layout? I don't know the answer, not sure if we need more context here or not, but we could specify like, "Reminder short form, shows/hides X links" to make it more obvious what this is.

File: packages/fxa-auth-server/lib/senders/emails/templates/subscriptionAccountReminderFirst/index.stories.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import { Meta } from '@storybook/html';
+import { subplatStoryWithProps } from '../../storybook-email';
+
+export default {
+  title: 'SubPlat Emails/Templates/subscriptionAccountReminderFirst',
+} as Meta;
+
+const createStory = subplatStoryWithProps(
+  'subscriptionAccountReminderFirst',
+  'Sent when a user needs to finish setting up their account.',
```
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/templates/subscriptionAccountReminderFirst/index.stories.ts:14]: I think this could use a little more detail too. I think this is sent whenever a user signs up through the subplat passwordless flow and ignores the other email you have up in a PR right, and then they're sent this reminder to finish setting up the account?

File: packages/fxa-auth-server/test/local/senders/email.js [hunk 5]
```diff
@@ -1127,38 +1127,38 @@ const TESTS = [
...
-      { test: 'include', expected: configHref('accountFinishSetupUrl', 'second-subscription-account-reminder', 'subscrition-account-create-email', 'email', 'product_name', 'token', 'product_id', 'flowId', 'flowBeginTime', 'deviceId') },
+      { test: 'include', expected: configHref('accountFinishSetupUrl', 'second-subscription-account-reminder', 'subscription-account-create-email', 'email', 'product_name', 'token', 'product_id', 'flowId', 'flowBeginTime', 'deviceId') },
       { test: 'notInclude', expected: 'utm_source=email' },
     ]],
     ['text', [
-      { test: 'include', expected: `Create a password:\n${configUrl('accountFinishSetupUrl', 'second-subscription-account-reminder', 'subscrition-account-create-email', 'email', 'product_name', 'token', 'product_id', 'flowId', 'flowBeginTime', 'deviceId')}` },
+      { test: 'include', expected: `Create a password:\n${configUrl('accountFinishSetupUrl', 'second-subscription-account-reminder', 'subscription-account-create-email', 'email', 'product_name', 'token', 'product_id', 'flowId', 'flowBeginTime', 'deviceId')}` },
```
Comment (LZoog) [inline packages/fxa-auth-server/test/local/senders/email.js:1161]: what, you don't think users should have subscritions?! 😉 Good catch.

Comment (xlisachan) [reply packages/fxa-auth-server/test/local/senders/email.js:1161]: idk, sounds buggy. 🐛 ... crit ... critter... ... I'll see myself out.

File: packages/fxa-auth-server/lib/senders/emails/layouts/subscription/en.ftl [hunk 6]
```diff
@@ -16,5 +17,9 @@ subplat-cancel-plaintext = { subplat-cancel }:
 subplat-reactivate = Reactivate subscription
 subplat-reactivate-plaintext = { subplat-reactivate }:
 subplat-update-billing = Update billing information
+subplat-privacy-policy = { -brand-mozilla } Privacy Policy
+subplat-privacy-policy-plaintext = { subplat-privacy-policy }:
+subplat-cloud-terms = { -brand-firefox } Cloud Terms of Service
```
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/layouts/subscription/en.ftl:22]: Oh I'm sorry I missed this in my first review. We currently have "Firefox Cloud" as a brand now. This should be `{ -brand-firefox-cloud } Terms of Service`

File: packages/fxa-auth-server/lib/senders/emails/layouts/subscription/index.stories.ts [hunk 7]
```diff
@@ -17,6 +17,13 @@ const createStory = subplatStoryWithProps(
   }
 );
 
+export const LayoutNoProduct = createStory(
+  {
+    reminderShortForm: true,
+  },
+  'Reminder short form - user has Firefox account, no specific product, shows Mozilla Privacy Policy and Firefox Cloud Terms of Service links, hides Terms and cancellation policy, Cancel/Reactivate subscription, and Update billing information links'
```
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/layouts/subscription/index.stories.ts:24]: I think it's OK if we want to leave this, but I think it might be harder to maintain/can easily become outdated and if we're going to be this specific, we should also be specific with the other subplat layout stories. You could also just shorten this to: `Reminder short form - no specified product` IMO to match the other stories.

Also, we were going to adjust the multiple product story at some point? I remember us bringing it up in a PR from last week or the week before but I can't remember what we said specifically (and I'm feeling lazy and not looking up my comment because it's after 6 😆)

**Review by LZoog — APPROVED** [review]
This is looking great, I could've blocked r+ on the "Firefox Cloud" bit but eh if you'll make that change and push this'll be g2g so I'll go ahead and approve. 🙂

**Comment by LZoog** [comment]
Oh also - will you make note of the changes in style in the confirmation needed doc? I think what you have is exactly correct because it's consistent, but it's a fairly big stylistic change and QA might have questions otherwise.

File: packages/fxa-auth-server/lib/senders/emails/layouts/subscription/en.ftl (hunk 6 above)
Comment (xlisachan) [reply packages/fxa-auth-server/lib/senders/emails/layouts/subscription/en.ftl:22]: omg I'm ashamed for not realizing Cloud Terms of Service is not a thing. 😂

File: packages/fxa-auth-server/lib/senders/emails/layouts/subscription/index.stories.ts (hunk 7 above)
Comment (xlisachan) [reply packages/fxa-auth-server/lib/senders/emails/layouts/subscription/index.stories.ts:24]: Updated reminder short form. 👍

Oh, I thought the `LayoutMultipleProducts` and its description of `Multiple products` sufficed. It was `LayoutNoProduct` before, which did not not make much sense. 👀
